chainer/link.py

```diff
@@ -416,8 +416,10 @@ def to_intel64(self):
             value = d[name]
             if isinstance(value, cuda.ndarray):
                 value = value.get()  # to numpy.ndarray
-            if (isinstance(value, numpy.ndarray) and
-                    intel64.inputs_all_ready((value,))):
+            if (isinstance(value, numpy.ndarray) and value.ndim in (1, 2, 4)):
```

Comment:
Why not use `intel64.inputs_all_ready`?

Reply:
`inputs_all_ready` is designed for testing if inputs are ready for iDeep primitives like `ideep4py.linear.Forward`. On the other hand, in this code we want to test if inputs can be converted to iDeep mdarray.

I think [`ideep4py.array`](https://github.com/intel/ideep/blob/bfdec56a9ecf21c85ae19c61e7c5b8554c56d12f/python/ideep4py/__init__.py#L35-L56) should be responsible for checking ndim is valid for mdarray, but such check is not implemented (python process dies if invalid ndim is passed to it).

Intel guy told me that iDeep is going to support arbitrary dimensions in the future version of iDeep. So, I'm considering this fix as a temporary workaround until iDeep is fixed.

Reply:
>`inputs_all_ready` is designed for testing if inputs are ready for iDeep primitives like `ideep4py.linear.Forward`.

OK. Please document it in `inputs_all_ready`.

>Intel guy told me that iDeep is going to support arbitrary dimensions in the future version of iDeep. So, I'm considering this fix as a temporary workaround until iDeep is fixed.

Also, please write a TODO about this.

Reply:
Thanks, fixed.